fix _unsafe_view composite compliance

bdhirsh opened the pull request:
`_unsafe_view` is listed as `CompositeExplicitAutograd`, but internally is calls `alias_with_sizes_and_strides` which isn't supported by non-eager backends.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77132
* #77131
* #77130
* #77129
* #77128
* __->__ #77127
* #77126
* #77125
* #76320
* #76126
* #76319

bdhirsh committed:
fix _unsafe_view composite compliance

[ghstack-poisoned]

albanD commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -2130,7 +2130,7 @@ inline Tensor view_impl(const Tensor& self, IntArrayRef size) {
     "not compatible with input tensor's size and stride (at least one dimension"
     " spans across two contiguous subspaces). Use .reshape(...) instead.");
-  return alias_with_sizes_and_strides(self, inferred_size, *stride);
+  return self.as_strided(inferred_size, *stride);
```
The whole point of `unsafe_view` is to be faster than doing a proper view that goes through the dispatcher.
So if we do this, we might as well remove `unsafe_view` altogether.

bdhirsh replied:
oh haha. Then maybe the `CompositeExplicitAutograd` label is just false advertising.

Ok, I'm actually going to just remove this PR. I guess we could replace `CompositeExplicitAutograd` with `CPU, CUDA, QuantizedCPU, QuantizedCUDA`, etc, but we risk breaking 3rd party backends that happened to be relying on it.

I have another PR later that will add a new alias key for basically "all backends except functional backends", which might be a good candidate for this kernel (since functional backends like XLA/LTC won't like the `alias_with_sizes_and_strides` call).

ezyang commented:
I'm wondering what the right thing to do here is. @albanD isn't the goal of `_unsafe_view` to bypass view metadata setup, not the dispatcher per se? If so we should make this a non-composite operator and just write the autograd function by hand

albanD commented:
Ho I forgot we added full autograd support for it. It is already going through the dispatcher once I gues (unless it is called via at::native).
But is the metadata update really that expensive compared to a full run through the dispatcher?

On a side note, isn't composite compliance only for the Implicit autograd keys?

ezyang commented:
> But is the metadata update really that expensive compared to a full run through the dispatcher?

Yes? Wasn't this the whole point of inference mode

bdhirsh commented:
> isn't composite compliance only for the Implicit autograd keys?

I'm probably overloading the term, yeah 😛. "Ops advertising as `CompositeExplicitAutograd` that don't actually work for all backends"

bdhirsh committed:
Update on "fix _unsafe_view composite compliance"

`_unsafe_view` is listed as `CompositeExplicitAutograd`, but internally is calls `alias_with_sizes_and_strides` which isn't supported by non-eager backends.




[ghstack-poisoned]

bdhirsh committed:
Update on "fix _unsafe_view composite compliance"

`_unsafe_view` is listed as `CompositeExplicitAutograd`, but internally is calls `alias_with_sizes_and_strides` which isn't supported by non-eager backends.




[ghstack-poisoned]
